streamduck-core/src/config.rs

```diff
@@ -544,6 +550,21 @@ impl DeviceConfig {
             _ => Kind::Original,
         }
     }
+
+    /// check if there are config changes
+    pub fn is_dirty(&self) -> bool {
+        self.dirty_state
+    }
+
+    /// remove dirty state
+    pub fn set_clean(&mut self) {
```

Comment:
Better terminology would be mark_clean, according to the industry anyways
```suggestion
    pub fn mark_clean(&mut self) {
```